app/controllers/password_resets_controller.rb

```diff
@@ -0,0 +1,35 @@
+class PasswordResetsController < ApplicationController
+
+  def new
+
+  end
+
+  def create 
+    @user = User.find_by(email: params[:email])
+      if @user.present?
+        #send email
+        PasswordMailer.with(user: @user).reset.deliver_now
+      end
+    redirect_to '/password/reset', notice: "If an account with that email was found, we have sent a link to reset your password"
+  end
+    
+  def edit
+    @user = User.find_signed!(params[:token], purpose: "password_reset")
```

Comment:
What does `find_signed!` do?

Reply:
it finds a record for particular period of time, here we set a time and purpose  if time is passed and purpose is not valid then it will not be valid. Here I am using it with token